src/core/processing/qgsprocessingalgrunnertask.cpp

```diff
@@ -23,7 +23,7 @@
 #include "qgsvectorlayer.h"
 
 QgsProcessingAlgRunnerTask::QgsProcessingAlgRunnerTask( const QgsProcessingAlgorithm *algorithm, const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
-  : QgsTask( tr( "Executing “%1”" ).arg( algorithm->displayName() ), QgsTask::CanCancel )
+  : QgsTask( tr( "Executing “%1”" ).arg( algorithm->displayName() ), algorithm->flags() & QgsProcessingAlgorithm::FlagNoThreading ? QgsTask::CanCancel : QgsTask::Flag() )
```

Comment:
Isn't this logic backwards?

Reply:
This looks strange for me too. Don't know why, but backwards (and correct) condition
```
algorithm->flags() & QgsProcessingAlgorithm::FlagNoThreading ? QgsTask::Flag() : QgsTask::CanCancel
```
does not work for me, for example, with GDAL algorithms I still have a cancel button in the task manager.

Reply:
Actually the flag tested is wrong - it should be  FlagCanCancel